db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SelectNode.java

```diff
@@ -2054,6 +2054,19 @@ public ResultSetNode modifyAccessPaths() throws StandardException {
                         getContextManager()
                         );
             }
+            /*
+             * During Parsing phase, we construct a HalfOuterJoinNode and if it is a right outer join,
+             * we set joinResultSet and rightOuterJoinUsingClause in every result column of top select nodes.
+             * However, here, we removed the original HalfOuterJoinNode to create a new one.
+             * The new one will get a resultSetNumber assigned to it.
+             * We need to clear the top SelectNode of its reference to the original HalfOuterJoinNode
+             */
+            for (ResultColumn rc : this.getResultColumns()) {
+                rc.setJoinResultset(null);
+                rc.setRightOuterJoinUsingClause(false);
+                joinNode.isJoinColumnForRightOuterJoin(rc);
+            }
+
```

Comment:
looks good to me, however i don't really understand why this would only apply to natural joins. to me it looks like natural join is a normal join without a ON clause and a generated USING clause. i think one could create a right outer join which would run into the same problem, right?

Reply:
See comments in `ResultColumn.rightOuterJoinUsingClause`. This was introduced by [DERBY-4631](https://issues.apache.org/jira/browse/DERBY-4631?attachmentSortBy%3DdateTime) and is only used for right outer join with using / natural